Common/Logger.swift

```diff
@@ -50,67 +58,97 @@ class Logger {
+            return
         }
         
+        // create log directory if doesn't already exist
         let fm = FileManager.default
-        guard let list = try? fm.contentsOfDirectory(at: appGroupURL, includingPropertiesForKeys: nil, options: []) else {
-            zLog.warn("Logger.cleanup Unable to search log directory for log files to clear")
+        let logDir = appGroupURL.appendingPathComponent("logs", isDirectory:true)
+        do {
+           try fm.createDirectory(at: logDir, withIntermediateDirectories: true)
+        } catch {
+            zLog.error("Error creating log dir \(error.localizedDescription)")
             return
         }
         
-        let now = Date()
+        // remove any logs > NLOGS
+        guard let list = try? fm.contentsOfDirectory(at: logDir, includingPropertiesForKeys: nil, options: []) else {
+            zLog.warn("Logger.cleanup Unable to search log dir for log files to clear")
+            return
+        }
         list.forEach { url in
-            if url.lastPathComponent.starts(with: tag) && url.pathExtension == "log" {
-                if let attrs = try? fm.attributesOfItem(atPath: url.path), let iat = attrs[.creationDate] as? Date {
-                    let daysOld = ((now.timeIntervalSince(iat) / 60) / 60) / 24
-                    if daysOld > 2.0 {
-                        zLog.info("Removing \(url.lastPathComponent) (is over \(Int(daysOld)) days old)")
-                        do { try fm.removeItem(at: url) }
-                        catch { zLog.error("Unable to remove \(url.lastPathComponent). Error:\(error.localizedDescription)") }
+            let fn = url.lastPathComponent
+            if fn.starts(with: tag) {
+                let comps = fn.components(separatedBy: ".")
+                if comps.count >= 3, let indxStr = comps.last, let indx = Int(indxStr) {
+                    if indx >= Logger.MAX_NUM_LOGS {
+                        do {
+                            try fm.removeItem(at: url)
+                        } catch {
+                            zLog.error("Unable to remove \(url.lastPathComponent). Error:\(error.localizedDescription)")
+                        }
                     }
-                } else {
-                    zLog.error("Logger.cleanup Unable to get file attributes pf \(url.lastPathComponent)")
+                }
+            }
+        }
+        
+        // remove "old style" logs from previous version that were stored in app group directory
+        guard let list = try? fm.contentsOfDirectory(at: appGroupURL, includingPropertiesForKeys: nil, options: []) else {
+            zLog.warn("Logger.cleanup Unable to search app directory for old-style log files to clear")
+            return
+        }
+        list.forEach { url in
+            if url.lastPathComponent.starts(with: "TUN_") ||  url.lastPathComponent.starts(with: "APP_") && url.pathExtension == "log" {
+                do {
+                    try fm.removeItem(at: url)
```

Comment:
Personally, I'd log this action in case someone reads this log and wonders where they went